source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1198,6 +1199,81 @@ TEST(NiFakeServiceTests, NiFakeService_GetAttributeViString_CallsGetAttributeViS
   EXPECT_THAT(response.attribute_value(), ElementsAreArray(attribute_char_array, expected_size));
 }
 
+TEST(NiFakeServiceTests, NiFakeService_GetViUInt8_CallsGetViUInt8)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViUInt8 a_ViUInt8_number = 255;
+  EXPECT_CALL(library, GetViUInt8(kTestViSession, _))
+      .WillOnce(DoAll(SetArgPointee<1>(a_ViUInt8_number), Return(kDriverSuccess)));
+
+  ::grpc::ServerContext context;
+  nifake_grpc::GetViUInt8Request request;
+  request.mutable_vi()->set_id(session_id);
+  nifake_grpc::GetViUInt8Response response;
+  ::grpc::Status status = service.GetViUInt8(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kDriverSuccess, response.status());
+  EXPECT_EQ(a_ViUInt8_number, response.a_uint8_number());
+}
+
+TEST(NiFakeServiceTests, NiFakeService_ViUInt8ArrayInputFunction_CallsViUInt8ArrayInputFunction)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViInt32 number_of_elements = 3;
+  ViUInt8 expected_array[] = {0, 255, -1};
```

Comment:
Assigning `-1` to an `unsigned char` passes in `Windows Build` but gives compilation error in `Ubuntu Build` and `Linux-RT Build`.
So should this be considered as a valid test case or should I use any non-negative value instead of -1?
Any thoughts @doshirohan ?

Reply:
@ni-pkhare , Datatype is ViUnt8 know. Why do we need to test with negative values? I think you can remove that -1. Same for other comment below

Reply:
@kruthi-NI , I wanted to check if it reads it as 255 or not. Thats it!

Reply:
We are not expecting that negative values will be passed for this. Please remove this.

Reply:
Done.

Reply:
Agree with Kruthi. While functionally it should work the way you have mentioned, passing in negative value for UInt8 is definitely not a valid use case and need not be tested.

Reply:
Yes, I have removed it.